tests/zfs-tests/tests/functional/libtest/libtest_faketty.ksh

```diff
@@ -0,0 +1,69 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2020 Adam Moss, Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# libtest's faketty should conform to expected semantics
+#
+# STRATEGY:
+# 1. Verify that faketty as subordinate works and correctly
+#      returns error code of sub-subordinates
+# 2. Verify that faketty without subordinates is pass-through
+# 3. Verify the faketty return code when used directly as function
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	# nothing special to do
+}
+log_onexit cleanup
+
+log_assert "libtest's faketty conforms to expected semantics"
+
+# NOTE: this is a tough set of tests for faketty, more strict than
+# expected by current users of faketty - could relax
+
+# 1.
+# `which tty` is weird, it's not finding the 'tty' util otherwise
```

Comment:
If you intend to use the `tty` command you'll need to add it to `tests/zfs-tests/include/commands.cfg` for it to be included in the bin dir used for the restricted test PATH.

Reply:
Ahh!  Thanks!  That was a mystery!